src/main/scala/com/fulcrumgenomics/vcf/FixVcfPhaseSet.scala

```diff
@@ -0,0 +1,250 @@
+      this.lastChrom = variant.chrom
+    }
+
+    variants += 1
+    val genotypes: Map[String, Genotype] = variant
+      .genotypes
+      .map { case (sampleName: String, genotype: Genotype) =>
+        // update the genotype
+        val result = updateGenotype(variant=variant, genotype=genotype)
+        // log the results
+        result match {
+          case NotPhasedWithPhaseSetValue(_)   =>
+            logger.warning(
+              "Genotype had a phase set but was unphased:" +
+                f"${variant.chrom}:${variant.pos}:${variant.id.getOrElse(".")} PS=${genotype("PS")}" +
+                "; consider r-running using `-x/--phase-genotypes-with-phase-set`"
+            )
+          case PhasedMissingPhaseSetValue(_) =>
+            logger.warning(
+              f"Genotype had no phase set but was phased: ${variant.chrom}:${variant.pos}:${variant.id.getOrElse(".")}"
+            )
+          case _                  => () // do nothing
+        }
+        this.descriptionCounter.count(result.description)
+        // return the (potentially) new/updated genotype
+        (sampleName, result.genotype)
+      }
+    variant.copy(genotypes=genotypes)
+  }
+
+  def updateGenotype(variant: Variant, genotype: Genotype): Result = {
+    this.genotypes += 1
+
+    (genotype.phased, genotype.get[Any]("PS")) match {
+      case (true, None)  => PhasedMissingPhaseSetValue(genotype) // phased but no phase set, what can we do?
+      case (false, None) => NotPhasedNoPhaseSetValue(genotype)   // not phased and no phase set, everything is fine
+      case (false, Some(_)) if !phaseGenotypesWithPhaseSet =>    // unphased and a phase set, but we do not want to convert it to phased
+        NotPhasedWithPhaseSetValue(genotype.copy(attrs=genotype.attrs.filterNot(_._1 == "PS"))) // remove the phase set for good measure
+      case (isPhased, Some(oldValue)) => // may or not be phased, but has a phase set
+        // Get the new phase set
+        val phaseSetToValue = this.phaseSetToPositionBySample(genotype.sample)
+        val oldValueString  = oldValue.toString
+        val newValue        = phaseSetToValue.getOrElse(oldValueString, {
+          phaseSetToValue(oldValueString) = variant.pos
+          variant.pos
+        })
+        // Build the new set of attributes
+        val phaseSetAttrs = {
+          if (keepOriginal) Map("PS" -> newValue, "OPS" -> oldValueString)
+          else Map("PS" -> newValue)
+        }
+        // update the genotype
+        val newGenotype = genotype.copy(attrs = genotype.attrs.filterNot(_._1 == "PS") ++ phaseSetAttrs, phased=true)
+        // return based on if we phased the variant and if we added/updated the phase set value
+        (isPhased, oldValue == newValue && isPhased) match {
+          case (true, true)   => Valid(newGenotype)
+          case (true, false)  => UpdatedPhaseSetValue(newGenotype)
+          case (false, true)  => UpdatedPhaseOnly(newGenotype)
```

Comment:
How can this be true?  Your match is on `isPhased` which must be false _and_ `oldValue == newValue && isPhased` which must then necessarily also be false.